### ISSUE TITLE
Cannot select all columns in a cross join

### ISSUE THREAD

**Issue by juandent** [fnc12/sqlite_orm#1056]
I can't get all the columns of a cross join like this one: (it does not compile)

```C++
storage.select(asterisk<Employee>(), asterisk<Department>, cross_join<Department>()); 
``` 

This one does not compile either although they are semantically equivalent:

```C++
storage.select(asterisk<Employee>(), asterisk<Department>());
``` 

The error messages are: 

> dev\statement_serializer.h(56,20): error C2064: term does not evaluate to a function taking 2 arguments

This one compiles but only gives Employee's columns (although it does the cross_join correctly):

```C++
storage.select(asterisk<Employee>(), cross_join<Department>());
``` 

BTW, this last one produces the correct SQL string: 

```SQL
SELECT * FROM "Emp" CROSS JOIN "Dept"
```

but does not return the 12 columns it should (the 8 from Employee and the 4 from Dept)

so there is something wrong with the bindings I think (it only returns the 8 cols from Employee)

**Comment by fnc12** [comment]
```c++
storage.select(asterisk<Employee>(), asterisk<Department>, cross_join<Department>()); 
```
is not correct cause it means `SELECT *, * CROSS JOIN ...`.

```c++
storage.select(asterisk<Employee>(), asterisk<Department>());
```
is also not correct cause it means `SELECT *, * FROM ...`.

```c++
storage.select(asterisk<Employee>(), cross_join<Department>());
```
this is a bug. It needs to be fixed. It is related to `statement_serializer` which doesn't count with select constraints `cross_join`. Let's leave this issue opened until the bug is fixed

**Comment by trueqbit** [comment]
This is fixed by PR #1143, and is very closely related to [Is it possible to write 2 or more asterisks in a select?](1106).

So it is possible to write
```c++
select(columns(asterisk<Employee>(), asterisk<Department>()))
```
or
```c++
select(columns(asterisk<Employee>(), asterisk<Department>()),
       from<Employee>(),
       cross_join<Department>())
```
